modules/swagger-codegen/src/main/resources/typescript-angular2/api.mustache

```diff
@@ -18,10 +18,10 @@ export class {{classname}} {
     protected basePath = '{{basePath}}';
     public defaultHeaders : Headers = new Headers();
 
-    constructor(protected http: Http, basePath: string) {
```

Comment:
Maybe we can make basePath [optional](https://angular.io/docs/ts/latest/guide/dependency-injection.html#!#optional-dependencies) and create an [opaque token](https://angular.io/docs/js/latest/api/core/OpaqueToken-class.html) and inject it based up on this token.